Fixed .pl domain support and added support for .ua/.cn/.io domains.

click0 opened the pull request:
Fixed .pl domain support and added support for .ua/.cn/.io domains.

click0 committed:
Fixing typo.

click0 committed:
Fixed .pl domain support.

click0 committed:
Added support for .ua/.cn/.io domains.

click0 committed:
Update domain-list.txt

click0 committed:
Fixed support for .jp/.aero/.cn/.pl/.md/.tr/.it/.mx domains.

click0 committed:
Changed the function of translating words into lowecase, because in some OS the tr utility does not work correctly.

hawkeye116477 committed:
Add renewal date for .pl domains

click0 committed:
Merge pull request #1 from hawkeye116477/master

Add renewal date for .pl domains

click0 committed:
Added support for .kz domains

hawkeye116477 committed:
Added support for .online domain

click0 committed:
Added support for .app/.top domains.
Fixed support for .io/.xyz/.me/.pl/.ro domains.

hawkeye116477 committed:
Added support for .bid/.ng/.site domains (#4)

* Added support for .bid/.ng domains

* Added support for .site domains

hawkeye116477 committed:
Added support for .icu domain

click0 committed:
Merge pull request #5 from hawkeye116477/master

Added support for .icu domain

hawkeye116477 committed:
Added support for .is/.cloud domains

click0 committed:
Merge pull request #6 from hawkeye116477/master

Added support for .is/.cloud domains

hawkeye116477 committed:
Few improvements

Added support for .systems domains and fixed status when expiration date isn't detected.

click0 committed:
Merge pull request #7 from hawkeye116477/master

Added support for .systems domain and fixed status when expiration date isn't detected

click0 committed:
Partial syntax fixes in the script.

click0 committed:
Added getmonth_number function to convert the month number into a short form for writing the month name

click0 committed:
Making hyphenations very long string for readability and editing.

click0 committed:
Partially removed extra disk read operations.

click0 committed:
Changed version of the script and added a description of the this changes.

hawkeye116477 committed:
Fixed typo

click0 committed:
Merge pull request #8 from hawkeye116477/master

Fixed typo

click0 committed:
Added option to display the version of the script.
Added option to show debug information when running script.

click0 committed:
Fixed documentation after adding two options.

hawkeye116477 committed:
Added support for .expert/.express domains

click0 committed:
Merge pull request #9 from hawkeye116477/master

Added support for .expert/.express domains

hawkeye116477 committed:
Fixed support for .ca domain

hawkeye116477 committed:
Change Canadian whois server

hawkeye116477 committed:
Added support for .space domain

hawkeye116477 committed:
Fix changelog

hawkeye116477 committed:
Cleaned up

click0 committed:
Merge pull request #10 from hawkeye116477/master

Fixed support for .ca domain & added support for .space domain

click0 committed:
Changed script URL, because the main repository has not been updated for a long time.
Small typo fix. Thanks @gowthamgts.

hawkeye116477 committed:
Added support for .co.pl domain

hawkeye116477 committed:
Fixed version variable

hawkeye116477 committed:
Fixed version variable

click0 committed:
Merge pull request #11 from hawkeye116477/master

Added support for .co.pl domain and fixed version variable

click0 committed:
Added support for the domain zone .укр.

hawkeye116477 committed:
Added support for .fun domain

click0 committed:
Merge pull request #12 from hawkeye116477/master

Added support for .fun domain

watermark committed:
add museum tld support

add museum tld support

click0 committed:
Merge pull request #13 from watermark/patch-1

add museum tld support

click0 committed:
The version of the script has been changed and the domain has been added to the test domain list.

click0 committed:
Added domains from new zones to the test domain list.

hawkeye116477 committed:
Added support for .live domain

click0 committed:
Merge pull request #14 from hawkeye116477/master

Added support for .live domain

hawkeye116477 committed:
Added support for .sk domain

hawkeye116477 committed:
Fixed displaying of some long registrars

hawkeye116477 committed:
Fixed support for .jp domain

hawkeye116477 committed:
Update date

hawkeye116477 committed:
Fixed support for .jp domain

click0 committed:
Merge pull request #16 from hawkeye116477/master

Added support for .sk domain and fixes

hawkeye116477 committed:
Added support for .club domain

click0 committed:
Merge pull request #17 from hawkeye116477/master

Added support for .club domain

click0 committed:
Removed duplicate condition.

click0 committed:
Partial syntax fixes in the script.
Updated version of the script.

tdulcet commented:
I was able to rewrite the main part of this script that outputs the table (about 600 lines of code) in just 70 lines! My script works with all the TLDs currently supported by this script, except `.kz`, which [does not have a whois server](https://github.com/nixcraft/domain-check-2/pull/25/commits/82456a6a4423f4d3aaf42f1030df1fad24308106). Since I did not hard code any TLDs it should automatically work with all the TLDs that have a [whois server](https://github.com/rfc1036/whois/blob/next/tld_serv_list) and provide the expiration date. Here is the output from my script for the [example](https://en.wikipedia.org/wiki/Example.com) domains and all the domains in the [domain-list.txt](https://github.com/click0/domain-check-2/blob/master/domain-list.txt) file:

![image](https://user-images.githubusercontent.com/15305150/59422477-4fc52e00-8d85-11e9-8a76-250e85d9174e.png)

I have been testing it the last few months as people have been adding support for additional TLDs to this script and it has worked for these TLDs without any changes. The code for my script is available [here](https://github.com/tdulcet/Remote-Servers-Status/blob/master/domains.sh) for testing. I incorporated it into my larger [Remote Servers Status Monitoring Script](https://github.com/tdulcet/Remote-Servers-Status), which also supports domain expiry reminder notifications.

Some differences with my script:
* Some TLDs do not provide the registrar. This script shows the registrant instead, which may be confusing for users. My script just says "Unknown".
* This script has a status column to indicate if the domain is "Valid", "Expiring" or "Expired". My script outputs the expiration date and days left in green, yellow or red to indicate that.
* This script only outputs the expiration date in a static format. My script outputs the expiration date and time in the current time zone and local format.
* My script typically does not require any changes as new TLDs are added or when existing TLDs change their expiration date format.
* My script has better error handling and will show any errors from the `whois` commands.
* I fixed all the bugs found by [ShellCheck](https://www.shellcheck.net/).
* If provided a domain-list.txt file, my script removes duplicate domains. The [domain-list.txt](https://github.com/click0/domain-check-2/blob/master/domain-list.txt) file provided in this repository currently lists `google.us` and `google.co` twice.

I was not able to provide the output from this script for comparison, since the domain file feature is currently broken in this PR.

I welcome any feedback on my script and feel free to use any of it in this PR. Please let me know if you find a TLD that should work with my script but does not.

hawkeye116477 commented:
@tdulcet It doesn't work for me, after run it displays `date: wrong date` in my language (Polish).

hawkeye116477 committed:
Added support for .stream domain

hawkeye116477 committed:
Fixed condition with whois

hawkeye116477 committed:
Added missing whois server for .stream domain

click0 committed:
Merge pull request #18 from hawkeye116477/master

Added support for .stream domain and small fixes

tdulcet commented:
@hawkeye116477 Try adding `export LC_ALL=en_US.UTF-8` to the top of the script. That should make the `date` commands input/output the dates in English.

Edit: I tested it again with the two new TLDs that were added to this script (https://github.com/nixcraft/domain-check-2/pull/25/commits/4c3b2fee985b8c35f7b23a3e4899dc3818a521ee and https://github.com/nixcraft/domain-check-2/pull/25/commits/ead6a678f434a1f27a24fbaf9fe629362b6ccb7c) and my script still works without any changes. Here is the updated screenshot from my script:
![image](https://user-images.githubusercontent.com/15305150/60161815-19d47080-97ad-11e9-8934-6396caa9798c.png)

click0 committed:
Fixed support for .jp domain

click0 committed:
Fixed support for .jp domain

click0 committed:
Added support for .xxx domain.

hawkeye116477 commented:
Ok, now works :smile:

click0 committed:
Added support for .id domain.

hawkeye116477 committed:
Fixed status when expiration date is wrongly detected

hawkeye116477 committed:
Added support for .today domain

hawkeye116477 committed:
Add line breaks

click0 committed:
Merge pull request #22 from hawkeye116477/master

Small improvements

click0 committed:
Added support for .game domain.

click0 committed:
Spaces, tabs and blank lines in the domain list file are now ignored.
Commenting on domains with the # symbol also began to work.

rk00t committed:
added support for .do domains

click0 committed:
Merge pull request #24 from rk00t/patch-do-domains

added support for .do domains

click0 committed:
Added support for .team/.app/.host/.website domains.

click0 committed:
Fixed length of registrar name for .edu domain.

click0 committed:
Merged some checks

click0 committed:
Added support for .com.br domain

click0 committed:
Added correct separator if time stamp contains 'T' character.

click0 committed:
Added an additional function to change to the desired date format

click0 committed:
Fixed display of the registrar in the .cn domain.

click0 committed:
Some variables are localized within only their function call.

click0 committed:
Added variables of two system programs.

click0 committed:
Added variable of one system program.

click0 committed:
Added support for .tg/.co.il/.net.il/.org.il domains.

click0 committed:
The version of the script has been changed.

click0 committed:
Changing the test domain in the .co.pl zone

click0 committed:
Remove leading spaces and tabs in the registrar name.
More precisely, we are looking for the Expiration domain from unknown domain zones.

click0 committed:
Added support for .am/.xin domain zones.

23ewrdtf committed:
Updated Readme to correct wget url

Updated wget URLs to point to the raw version of the file.

click0 committed:
Merge pull request #25 from tretos53/patch-1

Updated Readme to correct wget url

click0 committed:
Merge remote-tracking branch 'origin/master'

click0 committed:
Added support for .rs domain zone.
Fixed typos.

click0 committed:
Fixed typos.

click0 committed:
Added domains from new zones to the test domain list.

click0 committed:
The version of the script has been changed.

click0 committed:
Fixed regression for .hr domain zones and many zones from Africa.

click0 committed:
Fixed display of the registrar name in the .uk domain zone.
Partially added .gg domain zone, but it does not display the year.

sentabi committed:
fix for whois on jwhois client

jwhois client on Fedora using uppercase D on Expiration Date

click0 committed:
Merge pull request #26 from sentabi/patch-2

fix for .com .net on jwhois client
todo: bug whith custom whois server.
Variable WHOIS_SERVER not used.

click0 committed:
Merge remote-tracking branch 'origin/master'

click0 committed:
Added work with specific whois servers.
Fixed typos.
Test domains changed.

click0 committed:
Fixed typos

click0 committed:
Fixed regular expressions inside awk to find the required strings.

copenhaus commented:
hmm, sorry, new to github, how do I upload my modified code?  I've added and modified some TLDs.

click0 committed:
Added support for .kr/.hk/.pt/.sg domains.

click0 committed:
Fixed checking on the existence of binary files of the necessary programs for the script to work.

click0 committed:
Fixed support for .md/.id/.kz/.uk domains. Fixed typos.

click0 committed:
Fixed typos.

click0 committed:
Added data output in CSV format.

click0 committed:
Fixed partial support for .md domain. Fixed typos.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Create FUNDING.yml

click0 committed:
Update README.md

axelvf committed:
Added support for .ar domains.

Argentina domains

axelvf committed:
Fixed typo.

click0 committed:
Empty changes

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed formatting when parsing the registrar of the .ar domain zone.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed support for .md/.kz/.net.hk domains. Fixed typos.

click0 committed:
The test domains for the .co.pl domain zone has been changed.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Simplified checking for .xyz/.fi/.fr zones.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .cf zones.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Removed extra double spaces.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Simplified checking for .dk/.tr zones

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added description for release 2.60.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
The list of requirements has been corrected and added to the project description.
Added description for release 2.61.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .bm domain.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Improved documentation syntax.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Removed the check for the existence of the mail client binary if the script does not use mail notification.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Removed whitespace escape for awk.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed registrar output formatting for some domain zones.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Updating the script version to 2.64.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed description of changes in the script version to 2.64.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .co.jp domain.
Updated script version to 2.65.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fix typo.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .ee TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .pk TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .biz.ua TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .gov.uk TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Partial fix for .biz.ua TLDs support.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixes for .co.pl TLD.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added .icu/.ee/.pk/.gov.uk/.biz.ua domains from new zones to the test domain list.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Removing a section with a search for a large list of domains in connection with the transfer to a new place in previous commits.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
The division of local variables into two stages - the declaration of the variables themselves and the assignment of values to these variables.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Optimization to get rid of long lines in the code.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Code optimization with minor bug fixes.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .co.ua TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .pp.ua TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
For .tr TLD, trim the length of the registrar name.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Updated script version to 2.66.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed registrar check for .co.ua/.pp.ua TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .sn TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed support for .sn TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed support for .ee TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .st TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .th/.co.th TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
The test domains for the .sn/.st/.th/.co.th TLDs has been changed.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fix typo.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .by TLD.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .ps TLD.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Updated script version to 2.67.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added support for .kg TLD.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Removed whitespace escape for awk.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added a set of test domains for (automated) testing.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Added a test domain for .rs domain zone.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed support for .md domain.
The owner of the .md zone hid the data of the registrar from whois server.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed support for .sk/.bm TLDs.
Fixed the list of test domains for (automated) testing.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed support for .sg TLD.
Typo corrected.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed support for .kr TLD.

Signed-off-by: vlad11 <admin@support.od.ua>

click0 committed:
Fixed support for .md/.ps TLDs.

Signed-off-by: vlad11 <admin@support.od.ua>

lglerup committed:
Fixed support for .dk TLD without registrar info

click0 committed:
Fixed support for .id TLD.

Fixed support for .id TLD.
